src/form.js

```diff
@@ -802,7 +802,32 @@ define([
             }
         },
         changeMugType: function (mug, questionType) {
-            this.mugTypes.changeType(mug, questionType);
+            var _this = this;
+            _this.mugTypes.changeType(mug, questionType);
+            _this.setMugActions(mug);
+        },
+        setMugActions: function(mug) {
+            var _this = this,
+                tree = _this.vellum.data.core.$tree,
+                action_id = "add_choice";
+
+            if (mug.__className === 'Select' || mug.__className === 'MSelect') {
+                tree.jstree(true).add_action(mug.ufid, {
```

Comment:
Does https://github.com/dimagi/Vellum/pull/847/commits/2042113ca7e23288f5308bc86293c51baa364893 strike you as an improvement?  I'll take a stab at separating tree logic into its own file, but yeah, that might not be part of this PR.